File: admin-dash/Admin-Dash/Models/MetricExerciseJunction.cs
using System;
using System.Collections.Generic;

namespace Admin_Dash.Models;

public partial class MetricExerciseJunction
{
    public long Id { get; set; }

    public long MetricId { get; set; }

    public long ExerciseId { get; set; }

    public virtual Exercise Exercise { get; set; } = null!;

    public virtual MetricDefinition Metric { get; set; } = null!;
}

File: data-tool/src/main/java/com/nilecochen/fitnessapp/datatool/entities/MetricExerciseJunction.java
package com.nilecochen.fitnessapp.datatool.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "metric_exercise_junction", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"metric_id", "exercise_id"})
})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MetricExerciseJunction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "metric_id", nullable = false)
    private MetricDefinition metricDefinition;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exercise_id", nullable = false)
    private Exercise exercise;
}
